video/out/x11_common.c

```diff
@@ -416,12 +416,14 @@ static void xrandr_read(struct vo_x11_state *x11)
         XRRProviderResources *pr = XRRGetProviderResources(x11->display, x11->rootwin);
         for (int i = 0; i < pr->nproviders; i++) {
             XRRProviderInfo *info = XRRGetProviderInfo(x11->display, r, pr->providers[i]);
-            char *amd = strcasestr(info->name, "amd");
-            char *intel = strcasestr(info->name, "intel");
-            char *nvidia = strcasestr(info->name, "nvidia");
-            char *radeon = strcasestr(info->name, "radeon");
-            x11->has_mesa = (amd || intel || radeon) ? true : false;
-            x11->has_nvidia = nvidia ? true : false;
+            struct bstr provider_name = bstr0(info->name);
+            bstr_lower(provider_name);
```

Comment:
this would write to `info->name`, probably not intended
consider: `struct bstr provider_name = bstrdup(x11, bstr0(info->name)`